app/src/main/java/com/github/se/eventradar/model/repository/user/FirebaseUserRepository.kt

```diff
@@ -144,11 +151,24 @@ class FirebaseUserRepository(db: FirebaseFirestore = Firebase.firestore) : IUser
       uid: String,
       folderName: String
   ): Resource<Unit> {
-    return Resource.Success(Unit)
+    val storageRef = Firebase.storage.reference.child("$folderName/$uid")
+    return try {
+      storageRef.putFile(selectedImageUri).await()
+      Resource.Success(Unit)
+    } catch (e: Exception) {
+      Resource.Failure(e)
+    }
   }
 
   override suspend fun getImage(uid: String, folderName: String): Resource<String> {
-    return Resource.Success("To Be Implemented")
+    val storageRef =
+        Firebase.storage.reference.child("$folderName/$uid.png") // Maybe add a parameter?
+    return try {
+      val url = storageRef.downloadUrl.await().toString()
```

Comment:
similar to the comment above. Also, I would check if the download is successful before converting to string (this could lead to an unwanted exception).